sdk/cosmos/src/clients/permission_struct.rs

```diff
@@ -33,17 +33,15 @@ where
     }
 }
 
-impl<'a, C, D, USER> HasHyperClient for PermissionStruct<'a, C, D, USER>
+impl<'a, C, D, USER> HasHttpClient for PermissionStruct<'a, C, D, USER>
 where
     C: CosmosClient + Clone,
     D: DatabaseClient<C> + Clone,
     USER: UserClient<C, D> + Clone,
 {
     #[inline]
-    fn hyper_client(
-        &self,
-    ) -> &hyper::Client<hyper_rustls::HttpsConnector<hyper::client::HttpConnector>> {
-        self.user_client.hyper_client()
+    fn http_client(&self) -> &dyn HttpClient {
```

Comment:
I wonder if returning an `Arc<Box<dyn HttpClient>>` is a more ergonimic thing to do.